File: CommonFramework/Assets/CScripts/Network/KCPProtocol/KCPNetwork.cs
using UnityEngine;
using System.Text;
using System.Net;
using System.Net.Sockets;
public class KCPNetwork
{
	private IPEndPoint m_remoteEndPoint;
	private KCPSocket m_kcpSocket;
	private System.Action<ByteBuffer> m_actionReceive;

	public void SetReceiveAction(System.Action<ByteBuffer> actionReceive)
	{
		m_actionReceive = actionReceive;
	}

	private void ActionReceive(byte[] data)
	{
		ByteBuffer bytebuffer = new ByteBuffer();
		bytebuffer.WriteBytesWithoutLength(data);
		if(m_actionReceive != null)
		{
			m_actionReceive(bytebuffer);
		}
	}

	public void Init(uint kcpid, string remoteIP, int localPort, int remotePort)
	{
		IPAddress ip = IPAddress.Parse(remoteIP);
		m_remoteEndPoint = new IPEndPoint(ip, remotePort);
		m_kcpSocket = new KCPSocket();
		m_kcpSocket.Init(kcpid, remoteIP, localPort, remotePort,ActionReceive);
	}

	public void Send(byte[] data)
	{
		m_kcpSocket.Send(data);
	}

	public void Update()
	{
		m_kcpSocket.Update();
	}

	public void Dispose()
	{
		m_kcpSocket.Dispose();
	}
}

File: CommonFramework/Assets/Lua/LuaScript/SystemModule/Network/KCP/KCPNetwork.lua
---
--- Created by luzhuqiu.
--- DateTime: 2017/6/19 上午9:54
---
KCPNetwork = class()
function KCPNetwork:GetInstance()
    if self.m_Instance == nil then
        self.m_Instance = KCPNetwork.new()
    end
    return self.m_Instance
end

function KCPNetwork:Dispose()
    self.kcpSocket:Dispose()
    self.kcpSocket = nil
    self.actionReceive = nil
end

function KCPNetwork:Init(kcpid,remoteIP,localPort,remotePort,actionReceive)
    self.kcpSocket = KCPSocket.New()
    local funcRecv = function(byteBuffer)

        local luabuffer = byteBuffer:ToLuaBuffer()
        local data = DataPacket.UnPacket(luabuffer)

        local id = data[1]
        local length = data[2]
        local seq = data[3]
        local dataBytes = data[4]

        if self.actionReceive ~= nil then
            self.actionReceive(dataBytes)
        end
    end
    self.actionReceive = actionReceive
    self.kcpSocket:Init(kcpid, remoteIP, localPort, remotePort,funcRecv);
end

---传入byte字符串
function KCPNetwork:Send(msgID,seq,data)
    local dataSend = DataPacket.Packet(msgID,seq,data)
    self.kcpSocket:Send(dataSend)
end

function KCPNetwork:Update()
    self.kcpSocket:Update()
end
